pod.go

```diff
@@ -286,6 +286,20 @@ type Cmd struct {
 	Group string
 }
 
+// HardwareConfig describes VM resources configuration.
+type HardwareConfig struct {
+	// Related to processor resources
+	CPUs    string
+	Cores   string
+	Sockets string
+	Threads string
+
+	// Related to memory resources
+	MemSize  string
+	MemSlots string
+	MemMax   string
+}
+
```

Comment:
This is a `Resources` structure and it should not be so close to a qemu definition. It should be more high level:

```
type Resources struct {
        VCPUs int

        // Memory is the amount of available memory on mega bytes.
        Memory int
}
```

Then it is up to the selected hypervisor to convert that into some option/command.